src/Primitives/src/StringValues.cs

```diff
@@ -218,25 +218,41 @@ static string GetJoinedStringValueFromArray(string[] values)
 
         public string[] ToArray()
         {
-            return GetArrayValue() ?? Array.Empty<string>();
+            // The same idea as GetArrayValue with slightly different semantics: This method always returns a copy of the array (just like the IEnumerable.ToArray extension method).
+
+            // Take local copy of _values so type checks remain valid even if the StringValues is overwritten in memory.
+            var value = _values;
+            switch (value)
+            {
+                case string[] values:
+#if NETCOREAPP
+                    return values.AsSpan().ToArray();
+#else
+                    return (string[])values.Clone();
```

Comment:
Can do a similar thing for .NET Standard; with the `System.Memory` package (which looks to already be included in the .csproj)
```suggestion
                    return new ReadOnlySpan<string>(array).ToArray();
```

Reply:
Let me check if that makes a difference in performance, I'm curious and I still have my test playground open.

Reply:
Actually `.AsSpan()` exists as an extension method in that package, so you should just be able to remove the `#if`

Reply:
Interestingly `ReadOnlySpan` is consistently faster than `Span` in this regard.

> BenchmarkDotNet=v0.11.5, OS=Windows 10.0.17763.529 (1809/October2018Update/Redstone5)
> Intel Core i5-6440HQ CPU 2.60GHz (Skylake), 1 CPU, 4 logical and 4 physical cores
> .NET Core SDK=3.0.100-preview5-011568
>   [Host]     : .NET Core 3.0.0-preview5-27626-15 (CoreCLR 4.6.27622.75, CoreFX 4.700.19.22408), 64bit RyuJIT
>   DefaultJob : .NET Core 3.0.0-preview5-27626-15 (CoreCLR 4.6.27622.75, CoreFX 4.700.19.22408), 64bit RyuJIT


|              Method |     Mean |     Error |    StdDev |
|-------------------- |---------:|----------:|----------:|
|               Clone | 82.88 ns | 1.9200 ns | 2.2857 ns |
|         SpanToArray | 34.36 ns | 0.7804 ns | 0.7300 ns |
| ReadOnlySpanToArray | 27.20 ns | 0.6373 ns | 0.9342 ns |

Looking at their sources there is only one notable difference between their constructors, this is only present in `Span`:
```
if ((object) default (T) == null && array.GetType() != typeof (T[]))
          ThrowHelper.ThrowArrayTypeMismatchException();
```
That `GetType()` might slow down things a slight bit. Does not matter much, but I can then just use `ReadOnlySpan` for both platforms, right?

Reply:
At a guess the covariance of the array is important for the Span type because you can update the elements; so the types need to be an exact match. For the ReadOnlySpan since you can't update the elements it doesn't matter...

> Does not matter much, but I can then just use ReadOnlySpan for both platforms, right?

Yes